### ISSUE TITLE
Use HumanDuration to show snapshot creation date

### ISSUE THREAD

**Issue by jandubois** [rancher-sandbox/rancher-desktop#5575]
I find the full date format hard to parse at a glance:

```console
$ rdctl snapshot ls
ID            Name    Created
zc0u1cvlfu    foo     Fri, 22 Sep 2023 14:11:45 PDT
```

I much prefer the way Github displays timestamps less than a month as a duration, e.g. `about a minute ago`, `4 hours ago`, `two weeks ago`.

This is also done by the `docker` cli:

```console
$ docker images
REPOSITORY                                          TAG       IMAGE ID       CREATED      SIZE
nginx                                               latest    61395b4c586d   2 days ago   187MB
```

This is implemented in the [HumanDuration](https://pkg.go.dev/github.com/docker/go-units#HumanDuration) function.

I notice that Github switches to an abbreviated date format for durations over a month, e.g. `Aug 1` instead of `2 months ago`. I think that is an improvement over the pure `HumanDuration` format, but I guess it is a personal preference. I don't know if there is a go library for it.

I think we should use this format in `rdctl snapshot ls`, but also include it in the JSON output. The UI could then display the human format by default and provide the actual timestamp in a hovertip (again, like Github does it).

**Comment by jandubois** [comment]
BTW, and this maybe should be a separate issue, but I don't think including the "ID" in the human readable output is useful. I expect to be able to restore a snapshot by name, not by ID.

I would rather see a (potentially abbreviated) "Note" value in the output, like

```console
$ rdctl snapshot ls
NAME         CREATED      NOTE
my-snapshot  3 hours ago  Deployed my-app 1.234 and imported all...
```

The `NAME` column would automatically be resized to fit the longest snapshot name, and `CREATED` similarly to fit the longest date value. The `NOTE` column would be abbreviated to the terminal width when STDOUT is a TTY, or otherwise display the full note (with newlines replaced by spaces).

**Comment by ericpromislow** [comment]
See #5582 for the UI issue for this work.
